matplotlib/charts.py

```diff
@@ -0,0 +1,118 @@
+import locale
+import csv
+import itertools
+from matplotlib import pyplot as plt
+import numpy as np
+
+MY_FILE = input('What file should I use?')
+
+
+def parse(raw_file, delimiter):
+    """Parses a raw CSV file to a Json-line object"""
+
+    with open(raw_file, 'r', encoding='latin1') as opened_file:
+        csv_data = csv.reader(opened_file, delimiter=delimiter)
+        parsed_data = []
+        remaining_lines = itertools.islice(csv_data, 7, None)
+        fields = next(remaining_lines)
+        for row in remaining_lines:
+            parsed_data.append(dict(zip(fields, row)))
+    return parsed_data
+
+
+def sumup_and_join_values(dict1, dict2):
+    sum_up = dict1['Betrag'] + dict2['Betrag']
+    dict1['Buchung'] = '; '.join((
+        dict1['Buchung'],
+        dict2['Buchung']))
+    dict1['Valuta'] = '; '.join((
+        dict1['Valuta'],
+        dict2['Valuta']))
+    dict1['Buchungstext'] = '; '.join((
+        dict1['Buchungstext'],
+        dict2['Buchungstext']))
+    dict1['Verwendungszweck'] = '; '.join((
+        dict1['Verwendungszweck'],
+        dict2['Verwendungszweck']))
+    dict1['Betrag'] = sum_up
+    dict2['Remove'] = 'To be removed'
+    return dict1, dict2
+
+
+def sumup_rewe_spendings(data):
+    for dict1, dict2 in itertools.combinations(data, 2):
+        if ('rewe' in dict1['Auftraggeber/Empfänger'].lower() and
+                'rewe' in dict2['Auftraggeber/Empfänger'].lower()):
+            sumup_and_join_values(dict1, dict2)
+    return data
+
+
+def sumup_similar_spendings(data):
+    """Search in the column 'Auftraggeber/Empfanger'.
+    If there are more entries with the same name
+    sum up the values and remove the second element"""
```

Comment:
If I understood correctly, and you want one value summed. You could use `reduce`